src/tab.rs

```diff
@@ -4411,7 +4451,19 @@ impl Tab {
                     match output.send(message).await {
                         Ok(()) => {}
                         Err(err) => {
-                            log::warn!("failed to send thumbnail for {:?}: {}", path, err);
+                            log::warn!("failed to send thumbnail for {:?}: {}", &path, err);
+                        }
+                    }
+
+                    let total_size = calculate_dir_size(&path);
```

Comment:
I think you need a new subscription for this, it looks like this only runs when thumbnailing a file?

Reply:
Do you mind if I make some changes as I figure things out?

Reply:
I initially tried to create a dedicated subscription but had a number of problems with ownership which were later solved. However, I realised that every time this subscription is used, the size of the directory can be calculated. This subscription populates changes in the dialogue just where the size of the directory is to be shown, so in a way they are related. If that doesn't add up, I can try creating a separate subscription.

Reply:
Please see this commit: https://github.com/pop-os/cosmic-files/pull/656/commits/f1c535f5f7a322cab969d8623dd6def0d2d30b88